WalletWasabi/Stores/IndexStore.cs

```diff
@@ -357,6 +359,182 @@ public async Task RemoveAllNewerThanAsync(uint height)
+			{ 809982, [254, 4, 33, 82, 80] }, { 810573, [247, 22, 30, 63, 165] }, { 812427, [101, 1, 158, 216, 27] },
+			{ 814691, [93, 29, 193, 127, 181] }, { 815889, [243, 25, 88, 26, 187] }, { 816440, [22, 26, 17, 47, 217] },
+			{ 818755, [140, 35, 23, 229, 42] }, { 820358, [27, 21, 95, 49, 81] }, { 825577, [82, 21, 82, 182, 124] },
+			{ 829078, [176, 24, 167, 232, 238] }, { 833595, [22, 17, 81, 212, 20] }, { 834202, [118, 23, 10, 208, 72] },
+			{ 838327, [63, 27, 14, 126, 166] }, { 839255, [133, 20, 162, 147, 219] }, { 841188, [214, 2, 34, 204, 44] },
+			{ 846233, [95, 11, 155, 23, 200] }, { 848132, [10, 1, 4, 126, 139] }, { 848184, [206, 16, 76, 137, 226] },
+			{ 849339, [252, 7, 58, 6, 122] }, { 849806, [57, 12, 31, 77, 130] }, { 849936, [181, 11, 178, 186, 110] },
+			{ 852121, [61, 29, 23, 161, 129] }, { 854972, [106, 39, 132, 97, 32] }, { 856445, [93, 21, 63, 244, 201] },
+			{ 859032, [130, 5, 94, 15, 80] }, { 859496, [85, 28, 105, 76, 130] }, { 859606, [57, 9, 137, 188, 162] },
+			{ 860959, [223, 6, 46, 91, 210] }, { 861255, [143, 8, 75, 55, 124] }, { 497765, [145, 189, 205, 124, 42] },
+			{ 498265, [70, 112, 210, 211, 95] }, { 504757, [84, 148, 40, 145, 123] },
+			{ 504859, [5, 169, 16, 180, 192] }, { 505607, [46, 119, 96, 241, 203] },
+			{ 505669, [170, 162, 176, 20, 188] }, { 518084, [90, 38, 224, 175, 227] },
+			{ 522741, [240, 177, 159, 22, 181] }, { 522895, [37, 215, 137, 176, 242] },
+			{ 525723, [54, 161, 194, 247, 145] }, { 532611, [42, 162, 88, 33, 47] },
+			{ 537553, [61, 131, 140, 227, 28] }, { 538528, [38, 55, 197, 8, 188] }, { 540550, [0, 32, 139, 64, 52] },
+			{ 540790, [21, 107, 24, 95, 121] }, { 541664, [0, 59, 36, 226, 144] }, { 542989, [23, 181, 222, 242, 255] },
+			{ 543087, [6, 143, 120, 223, 25] }, { 547388, [240, 191, 223, 24, 234] },
+			{ 563624, [42, 57, 201, 244, 20] }, { 573295, [49, 252, 224, 43, 30] }, { 577296, [5, 172, 101, 232, 68] },
+			{ 580014, [50, 210, 138, 25, 237] }, { 580016, [54, 118, 24, 0, 141] }, { 582811, [199, 4, 200, 31, 65] },
+			{ 586012, [202, 40, 97, 14, 212] }, { 592050, [143, 162, 192, 19, 178] },
+			{ 594558, [127, 20, 190, 241, 148] }, { 596244, [80, 89, 123, 254, 39] },
+			{ 620433, [22, 181, 102, 201, 212] }, { 682045, [168, 189, 246, 39, 24] }, { 740866, [133, 23, 5, 53, 77] }
+			#endregion
+		};
+
+		// If corrupted it must be before 861657.
+		uint lastBatchToTest = (uint) Math.Min(IndexStorage.GetBestHeight(), lastHeightPotentiallyAffected) - batchSize + 1;
+		uint currentHeight = StartingFilters.GetStartingFilter(Network.Main).Header.Height;
+
+		if (lastBatchToTest < currentHeight)
+		{
+			// Index is fresh.
+			return;
+		}
+
+		while (true)
+		{
+			var batch = IndexStorage.Fetch(currentHeight, (int)batchSize).ToList();
+
+			// Almost all correct filters starts with byte "253".
+			// So a filter is affected if it doesn't start with "253" but ends with "253", so that the endian was reversed, except
```

Comment:
The first thing in a binary-serialized compact filter is the number of items (scripts) contained in the filter encoded as a `varint`. This means that those filters with a number of items in the range [254, 65534] start with 0xFD. However,

Reply:
With insight on how filters are actually build we can remove few hacks like the false positives by verifying the parameters and `deleteAllUnderHeight`